### DISCUSSION TITLE
Can this be run on Raspberry Pi 4 (armv7 architecture)?

### DISCUSSION THREAD

**Discussion by jonesnc** [jesec/rtorrent#25] [Q&A] [answered]
I have a raspberry pi 4 that I'd like to use to run this on. I see that the Dockerfile only checks for 64-bit architectures, but I'm wondering if I could modify the Dockerfile to also support armv7, which is the architecture of my raspberry pi 4.

Does that sound possible?

Here's the error I see when I try to build a docker image on my RPi4:

```
Step 5/27 : RUN apk --no-cache add     -Xhttps://dl-cdn.alpinelinux.org/alpine/edge/testing     bazel
 ---> Running in 1c8ff4e41fd0
fetch https://dl-cdn.alpinelinux.org/alpine/edge/testing/armv7/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/armv7/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/armv7/APKINDEX.tar.gz
ERROR: unable to select packages:
  bazel (no such package):
    required by: world[bazel]
The command '/bin/sh -c apk --no-cache add     -Xhttps://dl-cdn.alpinelinux.org/alpine/edge/testing     bazel' returned a non-zero code: 1
```

**Comment by jesec** [comment, marked as answer]
Not with the current form. Bazel (the build system) does not support armv7. 

You would have to use CMake, instead. 

A better solution is to switch to 64-bit. rpi4 does support 64-bit.
